src/v/iceberg/values_json.cc

```diff
@@ -0,0 +1,674 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "iceberg/values_json.h"
+
+#include "iceberg/json_utils.h"
+
+#include <absl/strings/numbers.h>
+#include <boost/iostreams/device/array.hpp>
+#include <boost/iostreams/stream.hpp>
+#include <boost/range/irange.hpp>
+
+#include <chrono>
+#include <iomanip>
+#include <ranges>
+#include <stdexcept>
+
+namespace {
+
+using namespace iceberg;
+
+// non copying input stream
+using ibufstream
+  = boost::iostreams::stream<boost::iostreams::basic_array_source<char>>;
+
+std::string_view
+parse_string_view(const json::Value& data, std::string_view context = "") {
+    if (!data.IsString()) {
+        throw std::invalid_argument(fmt::format(
+          "Expected JSON string for {}, got {}", context, data.GetType()));
+    }
+    return std::string_view{data.GetString(), data.GetStringLength()};
+}
+
+iobuf hex_str_to_iobuf(std::string_view str) {
+    if (str.size() & 0x01ul) {
```

Comment:
Very odd to see this instead of "the usual" `% 2` check. But maybe it is just me thinking in math rather than bits. The generated code is the same.

Reply:
ha, yeah idk where that habit comes from